index.js

```diff
@@ -1,206 +1,229 @@
-var assign = Object.assign || require('es6-object-assign').assign
-var memoize = require('memoizee')
+'use strict'
 
-var parseArgForType = function (type, arg) {
+const memoize = require('memoizee')
```

Comment:
For the `import` syntax you need to use `babel`. I could add `babel` if you want and also make the static properties defined at the end (like ArgType, ItemType etc) as class properties.